File: backend/app/schemas/catalog.py
"""Schemas para los catálogos (productos, plantas, customers)."""
from pydantic import BaseModel


# ── Planta ────────────────────────────────────────────────────────────────────
class PlantaCreate(BaseModel):
    codigo: str
    nombre: str
    ubicacion: str | None = None


class PlantaUpdate(BaseModel):
    nombre: str | None = None
    ubicacion: str | None = None


class PlantaOut(BaseModel):
    id: int
    codigo: str
    nombre: str
    ubicacion: str | None = None

    model_config = {"from_attributes": True}


# ── Customer ──────────────────────────────────────────────────────────────────
class CustomerCreate(BaseModel):
    codigo: str
    nombre: str


class CustomerUpdate(BaseModel):
    nombre: str | None = None


class CustomerOut(BaseModel):
    id: int
    codigo: str
    nombre: str

    model_config = {"from_attributes": True}


# ── Producto ──────────────────────────────────────────────────────────────────
class ProductoCreate(BaseModel):
    sku: str
    nombre: str
    unidad: str = "kg"
    familia: str | None = None
    categoria: str | None = None


class ProductoUpdate(BaseModel):
    nombre: str | None = None
    unidad: str | None = None
    familia: str | None = None
    categoria: str | None = None


class ProductoOut(BaseModel):
    id: int
    sku: str
    nombre: str
    unidad: str
    familia: str | None = None
    categoria: str | None = None

    model_config = {"from_attributes": True}

File: frontend/src/lib/api/catalog.ts
import { request } from './client'

export interface Planta {
  id: number
  codigo: string
  nombre: string
  ubicacion?: string | null
}
export interface Customer {
  id: number
  codigo: string
  nombre: string
}
export interface Producto {
  id: number
  sku: string
  nombre: string
  unidad: string
  familia?: string | null
  categoria?: string | null
}

const json = (body: unknown) => ({ method: 'POST', body: JSON.stringify(body) })
const patch = (body: unknown) => ({ method: 'PATCH', body: JSON.stringify(body) })
const del = { method: 'DELETE' }

export const catalogApi = {
  // Plantas
  plantas: () => request<Planta[]>('/catalog/plantas'),
  createPlanta: (p: Omit<Planta, 'id'>) => request<Planta>('/catalog/plantas', json(p)),
  updatePlanta: (id: number, p: Partial<Omit<Planta, 'id' | 'codigo'>>) => request<Planta>(`/catalog/plantas/${id}`, patch(p)),
  deletePlanta: (id: number) => request<void>(`/catalog/plantas/${id}`, del),

  // Customers
  customers: () => request<Customer[]>('/catalog/customers'),
  createCustomer: (c: Omit<Customer, 'id'>) => request<Customer>('/catalog/customers', json(c)),
  updateCustomer: (id: number, c: Partial<Omit<Customer, 'id' | 'codigo'>>) => request<Customer>(`/catalog/customers/${id}`, patch(c)),
  deleteCustomer: (id: number) => request<void>(`/catalog/customers/${id}`, del),

  // Productos
  productos: () => request<Producto[]>('/catalog/productos'),
  createProducto: (p: Omit<Producto, 'id'>) => request<Producto>('/catalog/productos', json(p)),
  updateProducto: (id: number, p: Partial<Omit<Producto, 'id' | 'sku'>>) => request<Producto>(`/catalog/productos/${id}`, patch(p)),
  deleteProducto: (id: number) => request<void>(`/catalog/productos/${id}`, del),
}
